topics/canister-based-wallet-architects.md

```diff
@@ -0,0 +1,82 @@
+### Context
+
+#### Common pattern
+
+Web3 DApp, for example, DEX or Marketplace, usually wants to access the user's wallet. The common pattern is to treat the user's address as a wallet address and use it to access his/her assets stored in the ledger. This pattern is widely used in various blockchain ecosystems, mostly used in Ethereum. DApps on IC use this pattern as well, for example, Sonic.ooo, Entrepot.app, icDex.io. And also current token standards are also affected by this pattern.
+
+It seems nothing wrong that developers choose this way.
+
+### Problems in general
+
+1. Fragmentation of wallet creation and assets management.
+2. Cross DApp wallet usage and applied to the industry-wide pattern.
+
+#### Problems illustration
+
+However, on IC, 2 problems remain unresolved. If the community is not realizing these problems. The liquidity of assets and interoperability between DApps will be severely limited. Less liquidity means fewer transactions( or data exchanging), and if the situation continues, the ecosystem will eventually collapse and fewer DApps will remain.
+
+1. The first problem is issued in https://github.com/dfinity/wg-identity-authentication/issues/5, generally, it's protocol level requirement and solution for global delegation and interoperability of user identity. We believe DFINITY will try its best to make this happens.
+
+2. The second problem is related to the first one, but it's the interaction pattern of the canister. [NNS wallet](https://nns.ic0.app) and [AstroX ME Wallet](https://app.astrox.me) use` principal<>delegation identity<>ledger` to create wallet instances for users. And developers try to access users' NNS or ME wallet from their DApp, and they want to use wallet identity to call their DApp backend canister because they assume that assets are stored in NNS or ME wallet. Because both NNS and ME uses session key in their frontend, and for security reason (sessionKey exposure), NNS and ME wallet can not provide such kind of function call.
```

Comment:
How would you reframe this problem as a requirement? I'd like to add it to the [list of interoperability requirements](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/interoperability.md#interoperability-of-services-using-user-identity).